src/OAuth2Form.tsx

```diff
@@ -123,13 +128,13 @@ export const OAuth2Form = () => {
   };
 
   // if user has already consented to this client request, no need to ask again
-  const getHasAlreadyConsented = (token: string) => {
-    if (token && !error) {
+  useEffect(() => {
```

Comment:
Wrap the function in useEffect and only call it when we have successfully fetched the profile.